icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -607,7 +600,7 @@ private SingleUnitImpl nextSingleUnit() {
                         break;
 
                     case TYPE_SIMPLE_UNIT:
-                        result.setSimpleUnit(token.getSimpleUnitIndex(), simpleUnits);
+                        result.setSimpleUnit(token.getSimpleUnitIndex(), UnitsData.getSimpleUnits());
```

Comment:
getSimpleUnits() looks slow, in which case this will make nextSingleUnit() and UnitsParser.parse() slow.

How about restoring `String[] simpleUnits` as static, and keeping the results in the class?

Reply:
why it is slow ? 
it will return just a reference to simpleUnits ??

Reply:
simpleUnits will be stored in UnitsData, it is not like the other data such as conversion rates.

Reply:
:facepalm: - looks like I wasn't reading thoroughly? I see this now, which I must have skipped last week, sorry:

        if (simpleUnits != null) {
            return simpleUnits;
        }

Reply:
Marking as resolved.